app/services/data_exports/transcription_file_generator.rb

```diff
@@ -0,0 +1,186 @@
+                   write_metadata_to_file,
+                   write_line_metadata_to_file)
+    end
+
+    private
+
+    # Private: creates raw data file as tempfile,
+    # returns tempfile
+    def write_raw_data_to_file
+      file = Tempfile.new(["raw_data_#{@transcription.id}-", '.json'])
+      file.write(@transcription.text)
+      file.rewind
+
+      file
+    end
+
+    # Private: creates consensus text file,
+    # returns location of the file
+    def write_consensus_text_to_file
+      file = Tempfile.new(["consensus_text_#{@transcription.id}-", '.txt'])
+      file.write(consensus_text)
+      file.rewind
+
+      file
+    end
+
+    # Private: retrieves and returns consensus text
+    def consensus_text
+      full_consensus_text = ""
+      @transcription.text.each do |key, value|
+        # if we find a frame, iterate through the lines of the frame
+        if /^frame/.match(key)
+          value.each do |line|
+            line_text = line['edited_consensus_text'].present? ?
+                        line['edited_consensus_text'] :
+                        line['consensus_text']
+            full_consensus_text.concat line_text + '\n'
+          end
+          # new line after every frame
+          full_consensus_text.concat '\n'
+        end
+      end
+
+      full_consensus_text
+    end
+
+    # Private: creates transcription metadata file,
+    # returns location of the file
+    def write_metadata_to_file
+      file = Tempfile.new(["transcription_metadata_#{@transcription.id}-", '.csv'])
+
+      CSV.open(file.path, 'wb') do |csv|
+        transcription_metadata.each do |csv_line|
+          csv << csv_line
+        end
+      end
+
+      file.rewind
+      file
+    end
+
+    # Private: creates transcription line metadata file,
+    # returns location of the file
+    def write_line_metadata_to_file
+      file = Tempfile.new(["transcription_line_metadata_#{@transcription.id}-", '.csv'])
+
+      CSV.open(file.path, 'wb') do |csv|
+        transcription_line_metadata.each do |csv_line|
+          csv << csv_line
+        end
+      end
+
+      file.rewind
+      file
+    end
+
+    # Private: retrieve and return transcription metadata formatted as
+    # array of csv lines
+    def transcription_metadata
+      csv_lines = []
+      csv_lines << [
+        'transcription id',
+        'internal id',
+        'reducer',
+        'caesar parameters',
+        'date approved',
+        'user who approved',
+        'text edited (T/F)',
+        'number of pages'
+      ]
+      csv_lines << [
+        @transcription.id,
+        @transcription.internal_id,
+        @transcription.reducer,
+        @transcription.parameters,
+        @transcription.updated_at,
+        @transcription.updated_by,
+        is_text_edited?,
+        @transcription.total_pages
+      ]
+    end
+
+    def is_text_edited?
+      # iterate through each 'frame' aka 'page' of transcription
+      @transcription.text.each do |key, lines|
```

Comment:
another way (more functional) approach for this is to filter the list of transcriptions searching for the ones that match the `frame` keys  using https://ruby-doc.org/core-2.7.0/Enumerable.html#method-i-filter

then you can apply your line tests on the resulting filtered collection.

Reply:
This is a good idea - didnt switch to use `.filter` in this case, but did refactor other loops in the PR to use it.